Packages/blueberry-jam-core/Runtime/LevelLoader/SceneTransitionerHelper.cs

```diff
@@ -0,0 +1,90 @@
+using System.Collections;
+using Codice.Client.BaseCommands.TubeClient;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace BJ
+{
+    internal class SceneTransitionerHelper : SingletonGameObject<SceneTransitionerHelper>
+    {
+        [Header("Game Object References")]
+        [SerializeField] private CanvasGroup Crossfade;
+        [SerializeField] private Image LoadingIcon;
+        [Header("Animation Reference Storage")]
+        [SerializeField] private Sprite[] Frames;
+
+        [Header("Animation Configuration")]
+        [SerializeField] private float TransitionTime = 1f;
+        [SerializeField] private AnimationCurve EaseTransitionCurve;
+
+        /* ----- Helper Variables -----*/
+        private float StartTransitionTime;
+
+        protected override void Awake()
+        {
+            base.Awake();
+            Assertions();
+
+            StartTransitionTime = 0.0f;
+        }
+
+        internal void LoadNewScene(string SceneName)
+        {
+            StartCoroutine(LoadLevelAnim(SceneName));
+        }
+
+        private IEnumerator LoadLevelAnim(string SceneName)
+        {
+
+            Assertions();
```

Comment:
I may be letting my C++ habits get the better of me here as well. Basically what I wanted to do was warn the user if they forgot to set any "config" variables in the prefab they are supposed to create (in this case it is SceneTransitionerConfig). Let me know what you think :)!

Reply:
I like the concept especially with the debug functions, I don't think we should just call it Assertions, it can be more specific.
Also, do we need to have/act on any return values from this function? It seems weird to just keep running as in I'm unsure how that wouldn't lead to a crash.

Reply:
We probably do not have to call it every frame, I have removed that.

I have changed the name to CheckSceneTransitionerManagerSettings to bake in semantic to the codebase